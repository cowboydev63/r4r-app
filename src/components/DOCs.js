import React from 'react';
import PropTypes from 'prop-types';
import { doCsInterface } from '../interfaces';
import { Theme } from '../theme';
import { renderDocsString } from '../utilities';

const DOCs = ({doCs}) => {
    if(!doCs.length) {
        return null;
    }
    return (
        <Theme element="article" className="resource__docs" aria-label="NCI Affiliation Information">
            <h2>NCI Affiliation</h2>
            { renderDocsString(doCs) }
        </Theme>
    );
}

DOCs.propTypes = {
    doCs: PropTypes.arrayOf(doCsInterface),
}

DOCs.defaultProps = {
    doCs: [],
}

export default DOCs;