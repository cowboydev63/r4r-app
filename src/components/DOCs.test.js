import React from 'react';
import { shallow } from 'enzyme';
import DOCs from './DOCs';

describe('DOCs Component', () => {
  it('renders correctly without poC', () => {
    const wrapper = shallow(<DOCs />);
    expect(wrapper).toMatchSnapshot();
})

it('renders correctly with poC', () => {
    const example = [{label: 'Eeinie'}]
    const wrapper = shallow(<DOCs doCs={ example }/>)
    expect(wrapper).toMatchSnapshot();
  })
})