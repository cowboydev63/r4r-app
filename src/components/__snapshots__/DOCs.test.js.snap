// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOCs Component renders correctly with poC 1`] = `
<Theme
  aria-label="NCI Affiliation Information"
  className="resource__docs"
  element="article"
>
  <h2>
    NCI Affiliation
  </h2>
  This resource is managed by the Eeinie.
</Theme>
`;

exports[`DOCs Component renders correctly without poC 1`] = `""`;
